observability/micrometer/src/main/java/org/wildfly/extension/micrometer/otlp/OtlpRegistryDefinitionRegistrar.java

```diff
@@ -0,0 +1,102 @@
+import org.wildfly.subsystem.resource.operation.ResourceOperationRuntimeHandler;
+import org.wildfly.subsystem.service.ResourceServiceConfigurator;
+import org.wildfly.subsystem.service.ResourceServiceInstaller;
+import org.wildfly.subsystem.service.capability.CapabilityServiceInstaller;
+
+public class OtlpRegistryDefinitionRegistrar implements ChildResourceDefinitionRegistrar, ResourceServiceConfigurator {
+    static final String NAME = "otlp";
+    public static final PathElement PATH = PathElement.pathElement("registry", NAME);
+
+    static final RuntimeCapability<Void> MICROMETER_OTLP_CONFIG_RUNTIME_CAPABILITY =
+            RuntimeCapability.Builder.of(MICROMETER_MODULE + ".wildfly-otlp-config", WildFlyMicrometerConfig.class)
+                    .build();
+
+    public static final SimpleAttributeDefinition ENDPOINT = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.ENDPOINT, ModelType.STRING)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    public static final SimpleAttributeDefinition STEP = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.STEP, ModelType.LONG, true)
+            .setDefaultValue(new ModelNode(TimeUnit.MINUTES.toSeconds(60)))
+            .setMeasurementUnit(MeasurementUnit.SECONDS)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    public static final Collection<AttributeDefinition> ATTRIBUTES = List.of(ENDPOINT, STEP);
+    private final WildFlyCompositeRegistry wildFlyRegistry;
+
+    public OtlpRegistryDefinitionRegistrar(WildFlyCompositeRegistry wildFlyRegistry) {
+
+        this.wildFlyRegistry = wildFlyRegistry;
+    }
+
+    @Override
+    public ManagementResourceRegistration register(ManagementResourceRegistration parent, ManagementResourceRegistrationContext context) {
+        ResourceRegistration registration = ResourceRegistration.of(PATH);
+
+        ResourceDescriptor descriptor = ResourceDescriptor.builder(MicrometerSubsystemRegistrar.RESOLVER.createChildResolver(PATH))
+                .addAttributes(ATTRIBUTES)
+                .withRuntimeHandler(ResourceOperationRuntimeHandler.configureService(this))
+                .build();
+        ManagementResourceRegistration mrr = parent.registerSubModel(
+                ResourceDefinition.builder(registration, descriptor.getResourceDescriptionResolver()).build());
+
+        ManagementResourceRegistrar.of(descriptor).register(mrr);
+
+        return mrr;
+    }
+
+    @Override
+    public ResourceServiceInstaller configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        String endpoint = OtlpRegistryDefinitionRegistrar.ENDPOINT.resolveModelAttribute(context, model).asStringOrNull();
+        Long step = OtlpRegistryDefinitionRegistrar.STEP.resolveModelAttribute(context, model).asLong();
```

Comment:
Nit: This statement won't result in null being assigned and in the end 'step' gets passed to Duration.ofSeconds(long), which would NPE if it was null. So using primitive long here and in WildFlyMicrometerConfig is clearer.

Reply:
Updated.